db/migrate/20150606065713_add_sti_to_lookup_keys.rb

```diff
@@ -0,0 +1,27 @@
+class AddStiToLookupKeys < ActiveRecord::Migration
+  def up
+    add_column :lookup_keys, :type, :string
+    LookupKey.scoped.each do |key|
+      if key.is_param
+        key.type = 'PuppetclassLookupKey'
+      else
+        key.type = 'VariableLookupKey'
+      end
+      key.save!(:validate => false)
+    end
+    add_index :lookup_keys, :type
+
+    remove_foreign_key :environment_classes, :lookup_keys if foreign_key_exists?(:environment_classes, :lookup_keys)
+    rename_column :environment_classes, :lookup_key_id, :puppetclass_lookup_key_id
+    add_foreign_key :environment_classes, :lookup_keys, :column => :puppetclass_lookup_key_id
+  end
+
+  def down
+    LookupKey.where(:type => "PuppetclassLookupKey").update_all(:is_param => true)
+    remove_column :lookup_keys, :type
+
+    remove_foreign_key :environment_classes, :lookup_keys if foreign_key_exists?(:environment_classes, :lookup_keys)
+    rename_column :environment_classes, :puppetclass_lookup_key_id, :lookup_key_id
+    add_foreign_key :environment_classes, :lookup_keys
```

Comment:
that should be the default name. the original definition uses that name because otherwise it would be env_classes_puppetclass_lk (I'm shortening because I'm lazy)